routes/users.js

```diff
@@ -1,9 +1,100 @@
-const express = require('express');
-const router = express.Router();
+const express = require('express')
+const { User, db } = require('../database/User.js')
+const router = express.Router()
+const mongoose = require('mongoose')
 
-/* GET users listing. */
-router.get('/', (request, response, next) => {
-  response.send('respond with a resource');
-});
 
-module.exports = router;
+router.get('/', ( request, response, next ) => {
+  response.send( 'respond with a resource' )
+})
+
+router.post('/', ( request, response, next ) => {
+  const { username } = request.body
+  const newUser = new User({
+    username: username,
+    dashboards: {
+       widgets: {}
+    }
+  })
+  newUser.save( ( error ) => {
+    if( error ) return handleError( error )
+  })
+})
+
+router.get('/widgets/:widgetId', ( request, response, next ) => {
```

Comment:
Please name the variable `:id` instead of `:widgetId` - given the name of the route, it is understood that this is a widget id.  Also, why is this route declared in the user routes file?  It should be in its own module for widgets route.